### PR TITLE
disable changing leading whitespace by default

### LINKED ISSUE #25: Please don't look at indent-tabs-mode
First of all, thanks for this definitely needed package!

Now, I'll try to explain my request:
- Personally, I do write code without tabs at all. I do have:
  
  ;; Indentation can't insert TABs.
  (setq-default indent-tabs-mode nil)
  
  in my Emacs configuration file.
- When editing someone else's code, they often use TABs to indent. In order to respect the spirit of the file, I do insert TABs explicitly (with `C-q TAB`)... but those get removed by ws-butler at save time...

Could you either never look at indent-tabs-mode, or add a variable to customize that?

Thanks a lot!

### PR DESCRIPTION
**Pull request by ian-kelling**
This fixes #25.

This should be merged instead of pull request #27. It's the same except condition on a config option instead of smart tabs (and use when instead of if).

### CODE REVIEW

**Comment by npostavs** [comment]
Perhaps also incorporate the check from on `smart-tabs-mode` from #27? Though it should be `(bound-and-true-p smart-tabs-mode)` to avoid error if `smart-tabs-mode` is not loaded.

**Comment by ian-kelling** [comment]
> Perhaps also incorporate the check from on smart-tabs-mode from #27? Though it should be (bound-and-true-p smart-tabs-mode) to avoid error if smart-tabs-mode is not loaded

I buy that. The pull request has been updated.

File: ws-butler.el [hunk 1]
```diff
@@ -78,6 +78,19 @@ i.e. only the \"virtual\" space is preserved in the buffer."
   :type 'boolean
   :group 'ws-butler)
 
+(defcustom ws-butler-convert-leading-tabs-or-spaces
+  t
+  "Make leading whitespace be tabs or spaces
+
+If `indent-tabs-mode' is non-nil, call `tabify', else call
+`untabify'. Do neither if `smart-tabs-mode' is enabled for this
+buffer. This is off by default because it's unwanted if you
```
Comment (npostavs) [inline ws-butler.el]: It appears to be on by default. Not sure whether you want to change the doc or the code...

Comment (ian-kelling) [inline ws-butler.el]: Oops. Was testing and forgot to change it back. I've updated it to be off.

Comment (lewang) [inline ws-butler.el]: Can we make the default value derive from smart-tabs-mode?

Comment (ian-kelling) [inline ws-butler.el]: > Can we make the default value derive from smart-tabs-mode?

No. smart-tabs-mode is only enabled in certain major modes, and I think that would be subject to which package got loaded first.

**Pull request merged**